pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTestBase.java

```diff
@@ -156,7 +157,11 @@ protected List<Object[]> queryH2(String sql)
     while (h2ResultSet.next()) {
       Object[] row = new Object[columnCount];
       for (int i = 0; i < columnCount; i++) {
-        row[i] = h2ResultSet.getObject(i + 1);
+        if (h2ResultSet.getMetaData().getColumnType(i + 1) == Types.OTHER) {
+          row[i] = h2ResultSet.getString(i + 1);
```

Comment:
This invokes the [method](http://javadox.com/com.h2database/h2/1.4.200/org/h2/value/ValueJson.html#getString()) to get the correct String data for `json` column data type.

Reply:
hmm I am not sure this is the right way to test b/c h2 doesn't have support for JSON. let alone the proper evaluation of json function results. 

instead we should use the `testQueryTestCasesWithOutput` method to not use H2 for these queries and directly verify the results

Reply:
H2 has the support [support](https://www.h2database.com/html/datatypes.html#json_type), but we can not test transformation functions with H2. I'll use your suggested one. Thanks